tests/components/demo/test_vacuum.py

```diff
@@ -47,326 +47,293 @@
+    assert state.state == STATE_ON
+
+    await common.async_start(hass, ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state == STATE_CLEANING
+
+    await common.async_pause(hass, ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state == STATE_PAUSED
+
+    await common.async_stop(hass, ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state == STATE_IDLE
+
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) < 100
+    assert state.state != STATE_DOCKED
+
+    await common.async_return_to_base(hass, ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state == STATE_RETURNING
+
+    await common.async_set_fan_speed(
+        hass, FAN_SPEEDS[-1], entity_id=ENTITY_VACUUM_STATE
+    )
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.attributes.get(ATTR_FAN_SPEED) == FAN_SPEEDS[-1]
+
+    await common.async_clean_spot(hass, entity_id=ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state == STATE_CLEANING
+
+
+async def test_unsupported_methods(hass):
+    """Test service calls for unsupported vacuums."""
+    hass.states.async_set(ENTITY_VACUUM_NONE, STATE_ON)
+    await hass.async_block_till_done()
+    assert vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    await common.async_turn_off(hass, ENTITY_VACUUM_NONE)
+    assert vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    await common.async_stop(hass, ENTITY_VACUUM_NONE)
+    assert vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    hass.states.async_set(ENTITY_VACUUM_NONE, STATE_OFF)
+    await hass.async_block_till_done()
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    await common.async_turn_on(hass, ENTITY_VACUUM_NONE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    await common.async_toggle(hass, ENTITY_VACUUM_NONE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    # Non supported methods:
+    await common.async_start_pause(hass, ENTITY_VACUUM_NONE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_NONE)
+
+    await common.async_locate(hass, ENTITY_VACUUM_NONE)
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_STATUS) is None
+
+    await common.async_return_to_base(hass, ENTITY_VACUUM_NONE)
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_STATUS) is None
+
+    await common.async_set_fan_speed(hass, FAN_SPEEDS[-1], entity_id=ENTITY_VACUUM_NONE)
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_FAN_SPEED) != FAN_SPEEDS[-1]
+
+    await common.async_clean_spot(hass, entity_id=ENTITY_VACUUM_BASIC)
+    state = hass.states.get(ENTITY_VACUUM_BASIC)
+    assert "spot" not in state.attributes.get(ATTR_STATUS)
+    assert state.state == STATE_OFF
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)